x-pack/plugins/endpoint/server/routes/resolver/queries/base.ts

```diff
@@ -7,50 +7,91 @@
 import { SearchResponse } from 'elasticsearch';
 import { IScopedClusterClient } from 'kibana/server';
 import { ResolverEvent } from '../../../../common/types';
-import {
-  paginate,
-  paginatedResults,
-  PaginationParams,
-  PaginatedResults,
-} from '../utils/pagination';
 import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
 import { legacyEventIndexPattern } from './legacy_event_index_pattern';
+import { MSearchQuery } from './multi_searcher';
 
-export abstract class ResolverQuery {
-  constructor(
-    private readonly indexPattern: string,
-    private readonly endpointID?: string,
-    private readonly pagination?: PaginationParams
-  ) {}
-
-  protected paginateBy(tiebreaker: string, aggregator: string) {
-    return (query: JsonObject) => {
-      if (!this.pagination) {
-        return query;
-      }
-      return paginate(this.pagination, tiebreaker, aggregator, query);
-    };
+/**
+ * ResolverQuery provides the base structure for queries to retrieve events when building a resolver graph.
+ *
+ * @param T the structured return type of a resolver query. This represents the type that is returned when translating
+ * Elasticsearch's SearchResponse<ResolverEvent> response.
+ */
+export abstract class ResolverQuery<T> implements MSearchQuery {
+  constructor(private readonly indexPattern: string, private readonly endpointID?: string) {}
+
+  private static createIdsArray(ids: string | string[]): string[] {
+    return Array.isArray(ids) ? ids : [ids];
   }
 
-  build(...ids: string[]) {
+  private buildQuery(ids: string | string[]): { query: JsonObject; index: string } {
+    const idsArray = ResolverQuery.createIdsArray(ids);
     if (this.endpointID) {
-      return this.legacyQuery(this.endpointID, ids, legacyEventIndexPattern);
+      return { query: this.legacyQuery(this.endpointID, idsArray), index: legacyEventIndexPattern };
     }
-    return this.query(ids, this.indexPattern);
+    return { query: this.query(idsArray), index: this.indexPattern };
+  }
+
+  private buildSearch(ids: string | string[]) {
+    const { query, index } = this.buildQuery(ids);
+    return {
+      body: query,
+      index,
+    };
+  }
+
+  protected static getResults(response: SearchResponse<ResolverEvent>): ResolverEvent[] {
+    return response.hits.hits.map(hit => hit._source);
   }
 
-  async search(client: IScopedClusterClient, ...ids: string[]) {
-    return this.postSearch(await client.callAsCurrentUser('search', this.build(...ids)));
+  /**
+   * Builds a multi search representation for this query
+   *
+   * @param ids a single more multiple (e.g. entity_id or unique_pid) to search for in the query
```

Comment:
:grey_question: This may be answered as I read along, but I'm wondering how it could be either an entity_id or a unique_pid
:grey_question: sp `more` -> 'or' I think

Reply:
I need to proof read my comments more haha. I'll expand the comment but, `entity_id` is for our new event types, and `unique_pid` is talking about our legacy endgame events.